File: src/application/dto/route.py
from pydantic import BaseModel, Field

from src.domain.enums import RouteTypeId, RouteTypeName


class RouteIdDTO(BaseModel):
    route_id: str = Field(description="Route ID")


class ConveyanceDTO(BaseModel):
    id: str = Field(description="Route ID")
    short_name: str = Field(description="Route short name")
    long_name: str = Field(description="Route long name")
    color: str = Field(description="Route color")
    type: RouteTypeId = Field(description="Route Type ID")
    type_name: RouteTypeName = Field(description="Route Type Name")

File: src/frontend/lib/application/dto/route.dart
class RouteTypeDTO {
    final String id;
    final String name;

    RouteTypeDTO({
        required this.id,
        required this.name,
    });

    factory RouteTypeDTO.fromJson(Map<String, dynamic> json) => RouteTypeDTO(
        id: json['id'],
        name: json['name'],
    );

    Map<String, dynamic> toQueryParameters() => {
        'id': id,
        'name': name,
    };
}

class RouteIdDTO {
    final String routeId;

    RouteIdDTO({
        required this.routeId,
    });

    factory RouteIdDTO.fromJson(Map<String, dynamic> json) => RouteIdDTO(
        routeId: json['route_id'],
    );

    Map<String, dynamic> toQueryParameters() => {
        'route_id': routeId.toString(),
    };
}

class ConveyanceDTO {
    final String id;
    final String shortName;
    final String longName;

    ConveyanceDTO({
        required this.id,
        required this.shortName,
        required this.longName,
    });

    String get displayName => "$shortName - $longName";

    factory ConveyanceDTO.fromJson(Map<String, dynamic> json) => ConveyanceDTO(
        id: json['id'],
        shortName: json['short_name'],
        longName: json['long_name'],
    );
}
